### ISSUE TITLE
Allow to speficy multiple lanes for a road.

### ISSUE THREAD

**Issue by ajh123** [Abelkrijgtalles/mojang-maps#5]
Some roads may be more complicate than others, for example:
![image](https://github.com/Abelkrijgtalles/mojang-maps/assets/41990982/996bdf5a-b999-4fea-b2ce-66f2f53aefc2)

Using a command like `/createroad Main_Street -129 -5 524 -129 -4 439 -129 -4 374 -133 -4 364 -137 -3 355 -139 -3 350 -139 -3 276 -129 -3 263 -129 -2 210` won't be enough because for some lanes the actionbar at the bottom believes your not on a road (because its too wide).

Maybe we can create an empty road like `/createroad Main_Street` then use `/addlane Main_Street TYPE -129 -5 524 -129 -4 439 -129 -4 374 -133 -4 364 -137 -3 355 -139 -3 350 -139 -3 276 -129 -3 263 -129 -2 210`. The `TYPE` parameter could be "Driving" or "Walking", it won't serve much use now, however, it will be beneficial to differentiate the types of a lane (could be useful for a Web UI in the future or other plugins that want to extend MojangMaps). In addition some servers may have a vehicle mod installed (like I do) so speficiying the type of the lane would mean you won't get people being told to walk in the middle of a road.

Using the hypothetical `/addlane` will mean we can define multiple lanes which we can class as the same road.

**Comment by Abelkrijgtalles** [comment]
This will be added in 1.6, but I don't have any idea when, because I kinda stopped developing Mojang Maps

**Comment by Abelkrijgtalles** [comment]
I have been thinking of the TYPE parameter for a long type, not in a way of lanes, but more in using multiple kinds/ways of transportation (horse, boat, elytra, etc.)

**Comment by ajh123** [comment]
> I have been thinking of the TYPE parameter for a long type, not in a way of lanes, but more in using multiple kinds/ways of transportation (horse, boat, elytra, etc.)

@Abelkrijgtalles That's basically the idea.

From earlier:
> The TYPE parameter could be "Driving" or "Walking"

however, more types would be better to. Maybe the types should be made by command / config files.

**Comment by Abelkrijgtalles** [comment]
And btw, I always make promises like: "This will be added 1.6" and now revisiting this issue out of curiosity, I don't know if this will really be added in 1.6, as it is a big feature, and most features are there own update. I've been thinking of going to version 2.x after features like particles, multiple ways of transportation and multi dimensional travel. This is because the naming scheme is not perfect and version 1.x a more buggy and beta version is. Version 2.x is really Mojang Maps as I imagined it when first starting.

**Comment by Abelkrijgtalles** [comment]
> > I have been thinking of the TYPE parameter for a long type, not in a way of lanes, but more in using multiple kinds/ways of transportation (horse, boat, elytra, etc.)
> 
> @Abelkrijgtalles That's basically the idea.
> 
> From earlier:
> 
> > The TYPE parameter could be "Driving" or "Walking"
> 
> however, more types would be better to. Maybe the types should be made by command / config files.

Yeah, that would be good, but only implementing the different ways through config would be difficult. What if it was a custom entity, how would you do that? This is a future problem, as I'm now trying to fix the minimizeJar conflict with javax.json.

**Comment by ajh123** [comment]
> > > I have been thinking of the TYPE parameter for a long type, not in a way of lanes, but more in using multiple kinds/ways of transportation (horse, boat, elytra, etc.)
> > 
> > 
> > @Abelkrijgtalles That's basically the idea.
> > From earlier:
> > > The TYPE parameter could be "Driving" or "Walking"
> > 
> > 
> > however, more types would be better to. Maybe the types should be made by command / config files.
> 
> Yeah, that would be good, but only implementing the different ways through config would be difficult. What if it was a custom entity, how would you do that? This is a future problem, as I'm now trying to fix the minimizeJar conflict with javax.json.

@Abelkrijgtalles 

I was thinking that the config wouldn't need to know if the transport is a custom entity, the config could just contain a list of names and how fast they are, like:
```yml
transport_types:
    horse:
        normal_speed: 0.4
    boat:
        normal_speed: 0.5
    rail:
        normal_speed: 1.0
```

**Comment by samuelh2005** [comment]
Actually I have a better idea.

We could follow what maps like Open Street Map does and replace "roads" with "ways", which "ways" are just a collection of "points"/"node".

Each node and way has a map of properties called [Tags](https://wiki.openstreetmap.org/wiki/Tags).

Doing this will mean we can store any information we want about an object.

**Comment by Abelkrijgtalles** [comment]
I already thought about using a ways system to use graphhopper core under the hood. I'll first be working on getting a prototype of #19 to work.

**Comment by Abelkrijgtalles** [comment]
See #20 for graphhopper core
